lib/src/main/kotlin/StatelessSolver.kt

```diff
@@ -239,6 +239,28 @@ class StatelessSolver<TState, TAction>(
         return optimalAction!!.parentAction ?: throw Exception("No action computed")
     }
 
+    private fun getNextOptimalActionInternal(node: StateActionNode<TAction>?): StateActionNode<TAction>? {
```

Comment:
There's an opportunity to simplify here. Instead of returning the best child StateActionNode, return the best action instead. This way getNextOptimalAction can simply call getNextOptimalActionInternal with the root node. getOptimalHorizon can also use this instead of resolving actions itself. 

Also, make the same improvement for the StatefulSolver?